### ISSUE TITLE
Support dynamic WS endpoint by using browserURL

### ISSUE THREAD

**Issue by shayke** [smooth-code/jest-puppeteer#210]
## 🐛 Bug Report

Puppeteer introduced a new `connect` property called `browserURL` which basically
does the dynamic WS endpoint fetch for you, see here: https://github.com/GoogleChrome/puppeteer/pull/3558
The problem is that the code adds the `browserWSEndpoint` no matter what config you give it, see: 
https://github.com/smooth-code/jest-puppeteer/blob/master/packages/jest-environment-puppeteer/src/PuppeteerEnvironment.js#L40

I'm running chrome in a docker container (https://hub.docker.com/r/justinribeiro/chrome-headless/) so basically my config is just:
```
connect: {
    browserURL: 'http://127.0.0.1:9222'
}
```

When running my tests, I'm getting this error:
```
TypeError: Cannot read property 'removeListener' of undefined
```

This happens because launching puppeteer fails and the exception is never catched so teardown happens:
```
Error: Exactly one of browserWSEndpoint, browserURL or transport must be passed to puppeteer.connect
```

**Comment by neoziro** [comment]
Hello @shayke, thanks for the report. Someone could help?

**Comment by xiaoyuhen** [comment]
will fix on this weekend

**Comment by lukaszfiszer** [comment]
Hi, author of changed that added `browserURL` to ppter here.

@shayke I cannot reproduce your error. The code at [src/PuppeteerEnvironment.js#L40](https://github.com/smooth-code/jest-puppeteer/blob/master/packages/jest-environment-puppeteer/src/PuppeteerEnvironment.js#L40) looks good to me as it is connecting using `browserWSEndpoint` to the instace of Chrome that was previously launched/connected to in [src/global.js](https://github.com/smooth-code/jest-puppeteer/blob/master/packages/jest-environment-puppeteer/src/global.js#L21) using. Can you paste here the complete config?

@xiaoyuhen IMHO your fix will not make the difference for the reasons described above

**Comment by xiaoyuhen** [comment]
@lukaszfiszer 

Unfortunately, I can't reproduce this problem, so it's hard to find the real problem.

**Comment by shayke** [comment]
I'm pretty sure this is now fixed due to #212 no?
I've opened this ticket before those changes in the code where the `browserWSEndpoint` was always provided to puppeteer.

**Comment by lukaszfiszer** [comment]
@shayke yeah, #212 definitely fixes this, so I guess @neoziro can now close this issue.
